retrieval/discovery/marathon/marathon.go

```diff
@@ -47,10 +48,32 @@ const appListPath string = "/v2/apps/?embed=apps.tasks"
 
 // Discovery provides service discovery based on a Marathon instance.
 type Discovery struct {
-	Servers         []string
-	RefreshInterval time.Duration
+	client          *http.Client
+	servers         []string
+	refreshInterval time.Duration
 	lastRefresh     map[string]*config.TargetGroup
-	Client          AppListClient
+	appsClient      AppListClient
+}
+
+// Initialize sets up the discovery for usage.
+func NewDiscovery(conf *config.MarathonSDConfig) (*Discovery, error) {
+	tls, err := httputil.NewTLSConfig(conf.TLSConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
```

Comment:
I'd use just the simple `Client.Timeout`. Good question on the default value. Maybe just stick that one line in with a fixed value of 30s for now (just so that there's a timeout at all) and make it configurable later? Also fine to do that all in another PR...